Library/Homebrew/search.rb

```diff
@@ -81,12 +81,14 @@ def self.search_casks(string_or_regex)
         end
       end
 
-      cask_tokens = Tap.flat_map(&:cask_tokens).filter_map do |c|
-        next if c.start_with?("homebrew/cask/") && !Homebrew::EnvConfig.no_install_from_api?
-
-        c.sub(%r{^homebrew/cask.*/}, "")
-      end
-      cask_tokens |= Homebrew::API::Cask.all_casks.keys unless Homebrew::EnvConfig.no_install_from_api?
+      cask_tokens = Tap.each_with_object([]) do |tap, array|
+        # We can exclude the core cask tap because `CoreCaskTap#cask_tokens` returns short names by default.
+        if tap.official? && !tap.core_cask_tap?
+          tap.cask_tokens.each { |token| array << token.sub(%r{^homebrew/cask.*/}, "") }
```

Comment:
I know it was there before, but what's the point of this sub?

Reply:
I'm guessing that we want to only search by cask token for internal taps so that `brew search version` doesn't return all `cask-versions` casks.

Reply:
Feels like this is kinda begging for something like `Formula#name` vs. `Formula#full_name`